usaspending_api/references/v2/views/cfda.py

```diff
@@ -0,0 +1,48 @@
+from rest_framework.response import Response
+from rest_framework.views import APIView
+from requests import post
+from time import sleep
+from django.conf import settings
+
+ALL_CFDAS = None
+
+
+class CFDAViewSet(APIView):
+    """
+    Return an agency name and active fy.
+    """
+
+    endpoint_doc = "usaspending_api/api_contracts/contracts/v2/references/agency/id.md"
+
+    def get(self, request, cfda=None):
+        """
+        Return the view's queryset.
+        """
+        self._populate_cfdas_if_needed()
+
+        if cfda:
+            results = [result for result in ALL_CFDAS if result["cfda"] == cfda]
```

Comment:
Since this is a GET request, shouldn't requesting a specific cfda which doesn't exist result in a 404 instead of an empty results response?

Reply:
If the endpoint was always returning a single result instead of a list, I would absolutely agree with you. With there being a list (that could have multiple values), I think there's a grey area. An alternative would be to have the endpoint not return a list when filtered down to one CFDA code (since this should only be getting one result anyway), then returning 404 if the CFDA isn't found, but that would be changing the contract underneath frontend mid-sprint, so I'd  want to talk to John first.

Reply:
There are two endpoints in this view, the responses don't need to be identical between them.

Can multiple CFDAs be returned when the client provides a specific CFDA `/api/v2/references/cfda/totals/{?cfda_code}/` ?

Reply:
It never should, since Grants API returns a JSON object instead of a JSON array, which guarantees unique keys. The only reason to keep this as a list is to avoid adding more work to @jonhill13 late in the sprint. John, do you forsee a problem if the api only returned results for a single CFDA instead of a results array when filtering on a single CFDA? I can get you a sample of what that would look like for you to develop against.

Reply:
Yes, i would recommend returning just an object when returning a specific result and if an error or any other message is returned from the grants api return that.

The only reason I wrote the contract like that was to make it easier on the backend, but in hindsight, I do not know why I did that. I guess trying to be nice to the backend but we should be following best practice @BrianZito @tony-sappe